File: app/src/main/java/ru/mamykin/foreignbooksreader/common/FolderToFilesListMapper.java
package ru.mamykin.foreignbooksreader.common;

import com.dropbox.core.v2.files.ListFolderResult;
import com.dropbox.core.v2.files.Metadata;

import java.util.ArrayList;
import java.util.List;

import ru.mamykin.foreignbooksreader.models.DropboxFile;
import rx.functions.Func1;

public class FolderToFilesListMapper implements Func1<ListFolderResult, List<DropboxFile>> {
    @Override
    public List<DropboxFile> call(ListFolderResult result) {
        List<Metadata> metadatasList = result.getEntries();
        List<DropboxFile> filesList = new ArrayList<>(metadatasList.size());
        for (Metadata metadata : metadatasList) {
            filesList.add(new DropboxFile(metadata, false));
        }
        return filesList;
    }
}

File: app/src/main/java/ru/mamykin/foreignbooksreader/common/FolderToFilesListMapper.kt
package ru.mamykin.foreignbooksreader.common

import com.dropbox.core.v2.files.ListFolderResult

import java.util.ArrayList

import ru.mamykin.foreignbooksreader.data.model.DropboxFile
import rx.functions.Func1

class FolderToFilesListMapper : Func1<ListFolderResult, List<DropboxFile>> {

    override fun call(result: ListFolderResult): List<DropboxFile> {
        val metaDataList = result.entries
        val filesList = ArrayList<DropboxFile>(metaDataList.size)
        for (metadata in metaDataList) {
            filesList.add(DropboxFile(metadata, false))
        }
        return filesList
    }
}